protozfitsreader/tests/test_read_raw_events.py

```diff
@@ -0,0 +1,292 @@
+    ]
+    expected_event_numbers = [
+        FIRST_EVENT_IN_EXAMPLE_FILE + i
+        for i in range(EVENTS_IN_EXAMPLE_FILE)
+    ]
+    assert event_numbers == expected_event_numbers
+
+
+def test_telescope_ids():
+    from protozfitsreader import ZFile
+    telescope_ids = [
+        event.telescope_id
+        for event in ZFile(example_file_path)
+    ]
+    expected_ids = [TELESCOPE_ID_IN_EXAMPLE_FILE] * EVENTS_IN_EXAMPLE_FILE
+    assert telescope_ids == expected_ids
+
+
+def test_num_gains():
+    from protozfitsreader import ZFile
+    num_gains = [
+        event.num_gains
+        for event in ZFile(example_file_path)
+    ]
+    expected_num_gains = [0] * EVENTS_IN_EXAMPLE_FILE
+    assert num_gains == expected_num_gains
+
+
+def test_n_pixel():
+    from protozfitsreader import ZFile
+    n_pixel = [
+        event.n_pixels
+        for event in ZFile(example_file_path)
+    ]
+    assert n_pixel == [1296] * EVENTS_IN_EXAMPLE_FILE
+
+
+def test_pixel_flags():
+    from protozfitsreader import ZFile
+    pixel_flags = [
+        event.pixel_flags
+        for event in ZFile(example_file_path)
+    ]
+    expected_pixel_flags = [
+        np.ones(1296, dtype=np.bool)
+    ] * EVENTS_IN_EXAMPLE_FILE
+
+    for actual, expected in zip(pixel_flags, expected_pixel_flags):
+        assert (actual == expected).all()
+
+
+def test_local_time():
+    from protozfitsreader import ZFile
+    local_time = [
+        event.local_time
+        for event in ZFile(example_file_path)
+    ]
+    assert local_time == EXPECTED_LOCAL_TIME
+
+
+def test_gps_time():
+    from protozfitsreader import ZFile
+    gps_time = [
+        event.central_event_gps_time
+        for event in ZFile(example_file_path)
+    ]
+    assert gps_time == EXPECTED_GPS_TIME
+
+
+def test_camera_event_type():
+    from protozfitsreader import ZFile
+    camera_event_type = [
+        event.camera_event_type
+        for event in ZFile(example_file_path)
+    ]
+    assert camera_event_type == [1, 1, 1, 1, 1, 8, 1, 1, 1, 1]
+
+
+def test_array_event_type():
+    from protozfitsreader import ZFile
+    array_event_type = [
+        event.array_event_type
+        for event in ZFile(example_file_path)
+    ]
+    assert array_event_type == [0] * EVENTS_IN_EXAMPLE_FILE
+
+
+def test_num_samples():
+    from protozfitsreader import ZFile
+    num_samples = [
+        event.num_samples
+        for event in ZFile(example_file_path)
+    ]
+    assert num_samples == [50] * EVENTS_IN_EXAMPLE_FILE
+
+
+def test_adc_samples():
+    from protozfitsreader import ZFile
+    adc_samples = [
+        event.adc_samples
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in adc_samples:
+        assert actual.dtype == np.int16
+        assert actual.shape == (1296, 50)
+
+    adc_samples = np.array(adc_samples)
+
+    # these are 12 bit ADC values, so the range must
+    # can at least be asserted
+    assert adc_samples.min() == 0
```

Comment:
is this always true ? I am confused should we use : lower or equal and greater or equal ?

Reply:
I think you are right

Reply:
This works only by chance

Reply:
done in https://github.com/cta-sst-1m/digicampipe/pull/104